### PR TITLE
 fix(VideoLoader): add playsinline attr to video element

### PR DESCRIPTION
**Pull request by happyhj**
## Issue
#143 

## Details
* reduce size of equi image to 4096 at manual test
* add unit tests
* fix karma config to prevent errors by including image file awith script tag

### CODE REVIEW

File: src/PanoImageRenderer/VideoLoader.js [hunk 1]
```diff
@@ -71,14 +78,18 @@ export default class VideoLoader {
 
 	get() {
 		return new Promise((res, rej) => {
+			// on iOS safari, 'loadeddata' will not triggered unless the user hits play,
+			// so used 'loadedmetadata' instead.
+			const isIOS = Agent().os.name === "ios";
+			const thresholdReadyState = isIOS ? READY_STATUS.HAVE_METADATA : READY_STATUS.HAVE_CURRENT_DATA;
+			const thresholdEventName = isIOS ? "loadedmetadata" : "loadeddata";
+
```
Comment (jongmoon) [inline src/PanoImageRenderer/VideoLoader.js]: @happyhj Is it necessary to branch iOS code? 
If necessary then... how about to move this code on VideoLoader constructor?

File: test/unit/YawPitchControl/DeviceMotion.spec.js [hunk 2]
```diff
@@ -100,6 +115,41 @@ describe("DeviceMotion", function() {
 			});
 		});
 
+		it("should not trigger devicemotion event on android", (done) => {
```
Comment (jongmoon) [inline test/unit/YawPitchControl/DeviceMotion.spec.js]: @happyhj devicemotion is not triggered on android?

Following `inst.disable()` call is the reason for not firing event. Isn't it?

File: src/PanoImageRenderer/VideoLoader.js (hunk 1 above)
Comment (happyhj) [reply src/PanoImageRenderer/VideoLoader.js]: I removed conditional logic.  VideoLoader will use HAVE_METADATA ready state to determine the video resource is resolved.

File: test/unit/YawPitchControl/DeviceMotion.spec.js (hunk 2 above)
Comment (happyhj) [reply test/unit/YawPitchControl/DeviceMotion.spec.js]: I agree with the description is not clear.
I will update the test description like below

> should not trigger "devicemotion" event on android after calling disable()

File: karma.conf.js [hunk 3]
```diff
@@ -47,7 +49,7 @@ module.exports = function(config) {
     customLaunchers: {
       ChromeHeadlessGL: {
         base: 'ChromeHeadless',
-        flags: ['--disable-gpu', '--use-gl=osmesa']
+        flags: ['--disable-gpu', '--use-gl=osmesa', '--no-sandbox', '--disable-setuid-sandbox']
```
Comment (jongmoon) [inline karma.conf.js]: @happyhj What is this?

File: src/PanoImageRenderer/VideoLoader.js [hunk 4]
```diff
@@ -40,45 +68,88 @@ export default class VideoLoader {
...
+			this._video.setAttribute("playsinline", "");
 
 			if (video instanceof Array) {
 				video.forEach(v => this._appendSourceElement(v));
 			} else {
 				this._appendSourceElement(video);
 			}
+		}
+
+		if (this._video) {
```
Comment (jongmoon) [inline src/PanoImageRenderer/VideoLoader.js]: @happyhj This condition is unnecessary.

File: src/PanoImageRenderer/VideoLoader.js [hunk 5]
```diff
@@ -40,45 +68,88 @@ export default class VideoLoader {
...
+		});
+	}
+
 	get() {
 		return new Promise((res, rej) => {
 			if (!this._video) {
 				rej("VideoLoader: video is undefined");
-			} else if (this._video.readyState >= READY_STATUS.HAVE_CURRENT_DATA) {
+			} else if (this._loadStatus === READY_STATUS.LOADING_FAILED) {
+				rej("VideoLoader: video source does not exist");
```
Comment (jongmoon) [inline src/PanoImageRenderer/VideoLoader.js]: @happyhj I think `video source is invalid.` is more proper message. How about you?

File: src/PanoImageRenderer/VideoLoader.js [hunk 6]
```diff
@@ -40,45 +68,88 @@ export default class VideoLoader {
...
 				res(this._video);
 			} else {
-				this._once("loadeddata", () => res(this._video));
-				// DO NOT HANDLE ERRORS, DELEGATE IT TO USER BY USING VIDEO ELEMENT.
-				// this._once("error", e => rej(`VideoLoader: failed to load ${e.target.src}`));
+				// check errorCnt and rejectx
+				const rejector = () => {
+					if (this._loadStatus === READY_STATUS.LOADING_FAILED) {
+						this._detachErrorHandler(rejector);
+						this._detachErrorHandler(this._onerror);
```
Comment (jongmoon) [inline src/PanoImageRenderer/VideoLoader.js]: Following is duplicated
```js
this._detachErrorHandler(this._onerror);
```

with detacher on `_onerror`.

File: test/unit/PanoImageRenderer/CubeRenderer.spec.js [hunk 7]
```diff
@@ -245,7 +245,7 @@ describe("CubeRenderer", () => {
 						done();
 					});
 				};
-			});
+			}).timeout(20000);
```
[the remark this answers is not in the document]
Comment (happyhj) [reply test/unit/PanoImageRenderer/CubeRenderer.spec.js]: I think we don't need timeout anymore. 
will remove that.

File: src/PanoImageRenderer/VideoLoader.js (hunk 5 above)
Comment (happyhj) [reply src/PanoImageRenderer/VideoLoader.js]: I also prefer `video source is invalid` because it is not a definite expression.

**Comment by jongmoon** [comment]
LGTM

**Pull request merged**